src/nl/hannahsten/texifyidea/completion/LatexBibliographyReferenceProvider.kt

```diff
@@ -0,0 +1,37 @@
+package nl.hannahsten.texifyidea.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionProvider
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.completion.impl.CamelHumpMatcher
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.util.ProcessingContext
+import nl.hannahsten.texifyidea.TexifyIcons
+import nl.hannahsten.texifyidea.completion.handlers.LatexReferenceInsertHandler
+import nl.hannahsten.texifyidea.psi.BibtexEntry
+import nl.hannahsten.texifyidea.util.findBibtexItems
+import java.util.*
+
+object LatexBibliographyReferenceProvider : CompletionProvider<CompletionParameters>() {
+    override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, result: CompletionResultSet) {
+        val lookupItems = parameters.originalFile.findBibtexItems()
+                .map { bibtexEntry ->
+                    val entry = bibtexEntry as BibtexEntry
```

Comment:
@fberlakovich There was one small mistake here: the bibtexEntry you get may actually not be a BibtexEntry but a LatexCommands, if the user defined the bibliography entry using the `\bibitem` command for example. (Resulted in #1229) The original LatexLabelReference did take this into account, so I've already fixed it.

Reply:
Ah, now I understand why 😃 , thanks for fixing it!